File: multitypelistviewadapter/src/main/java/net/kmfish/multitypelistviewadapter/IArrayAdapter.java
package net.kmfish.multitypelistviewadapter;

import java.util.List;

/**
 * Created by lijun on 16/8/5.
 */
public interface IArrayAdapter {

    void addData(Data item);

    void addDatas(List<Data> items);

    void addDatas(Data... items);

    void insert(Data item, int index);

    void remove(Data object);

    void remove(int pos);

    void clear();

    List<Data> getDatas();

    void setNotifyOnChange(boolean notifyOnChange);

    void notifyDataSetChanged();

    Data getData(int position);

    int getItemType(int position);

    int getCount();
}

File: multitypelistviewadapter/src/main/java/net/kmfish/multitypelistviewadapter/IArrayAdapter.kt
package net.kmfish.multitypelistviewadapter

/**
 * Created by lijun on 16/8/5.
 */
interface IArrayAdapter {

    fun addData(item: Data<*>)

    fun addDatas(items: List<Data<*>>)

    fun addDatas(vararg items: Data<*>)

    fun insert(item: Data<*>, index: Int)

    fun remove(item: Data<*>)

    fun remove(pos: Int)

    fun clear()

    fun setNotifyOnChange(notifyOnChange: Boolean)

    fun notifyDataSetChanged()

    fun getData(position: Int): Data<*>?
}
